diff -Nru curl-7.41.0/include/curl/curl.h curl-7.41.0-mod/include/curl/curl.h
--- curl-7.41.0/include/curl/curl.h	2015-02-25 05:49:02.000000000 +0800
+++ curl-7.41.0-mod/include/curl/curl.h	2016-09-16 15:54:28.679764086 +0800
@@ -383,6 +383,7 @@
   CURLINFO_DATA_OUT,     /* 4 */
   CURLINFO_SSL_DATA_IN,  /* 5 */
   CURLINFO_SSL_DATA_OUT, /* 6 */
+  CURLINFO_ERROR,        /* 7 */
   CURLINFO_END
 } curl_infotype;
 
diff -Nru curl-7.41.0/lib/sendf.c curl-7.41.0-mod/lib/sendf.c
--- curl-7.41.0/lib/sendf.c	2015-02-03 17:51:41.000000000 +0800
+++ curl-7.41.0-mod/lib/sendf.c	2016-09-16 15:56:56.083769553 +0800
@@ -161,7 +161,8 @@
       data->state.buffer[len] = '\n';
       data->state.buffer[++len] = '\0';
     }
-    Curl_debug(data, CURLINFO_TEXT, data->state.buffer, len, NULL);
+    /*Curl_debug(data, CURLINFO_TEXT, data->state.buffer, len, NULL);*/
+    Curl_debug(data, CURLINFO_ERROR, data->state.buffer, len, NULL);
   }
 
   va_end(ap);
@@ -659,6 +660,7 @@
   case CURLINFO_TEXT:
   case CURLINFO_HEADER_OUT:
   case CURLINFO_HEADER_IN:
+  case CURLINFO_ERROR:
     fwrite(s_infotype[type], 2, 1, data->set.err);
     fwrite(ptr, size, 1, data->set.err);
 #ifdef CURL_DOES_CONVERSIONS
